src/ChannelServer/World/Character.cs

```diff
@@ -306,251 +306,1830 @@ public int ClassLevel
+		/// Gets or sets character's Job vitality (CON).
+		/// </summary>
+		public int ConByJob { get; set; }
+
+		/// <summary>
+		/// Gets or sets character's Invested vitality (CON).
+		/// </summary>
+		public int ConInvested { get; set; }
+
+		/// <summary>
+		/// Gets or sets character's Item Bonus vitality (CON).
+		/// </summary>
+		public int ConBonus { get; set; }
+
+		/// <summary>
+		/// Gets character's vitality (CON).
+		/// </summary>
+		public int Con { get { return this.ConByJob + this.ConInvested + this.ConBonus; } }
+
+		/// <summary>
+		/// Gets or sets character's Job intelligence (INT).
+		/// </summary>
+		public int IntByJob { get; set; }
+
+		/// <summary>
+		/// Gets or sets character's Invested intelligence (INT).
+		/// </summary>
+		public int IntInvested { get; set; }
+
+		/// <summary>
+		/// Gets or sets character's Item Bonus intelligence (INT).
+		/// </summary>
+		public int IntBonus { get; set; }
+
+		/// <summary>
+		/// Gets character's intelligence (INT).
+		/// </summary>
+		public int Int { get { return this.IntByJob + this.IntInvested + this.IntBonus; } }
+
+		/// <summary>
+		/// Gets or sets character's Job spirit (SPR/MNA).
+		/// </summary>
+		public int SprByJob { get; set; }
+
+		/// <summary>
+		/// Gets or sets character's Invested spirit (SPR/MNA).
+		/// </summary>
+		public int SprInvested { get; set; }
+
+		/// <summary>
+		/// Gets or sets character's Item Bonus spirit (SPR/MNA).
+		/// </summary>
+		public int SprBonus { get; set; }
+
+		/// <summary>
+		/// Gets character's spirit (SPR/MNA).
+		/// </summary>
+		public int Spr { get { return this.SprByJob + this.SprInvested + this.SprBonus; } }
+
+		/// <summary>
+		/// Gets or sets character's Job agility (DEX).
+		/// </summary>
+		public int DexByJob { get; set; }
+
+		/// <summary>
+		/// Gets or sets character's Invested agility (DEX).
+		/// </summary>
+		public int DexInvested { get; set; }
+
+		/// <summary>
+		/// Gets or sets character's Item Bonus agility (DEX).
+		/// </summary>
+		public int DexBonus { get; set; }
+
+		/// <summary>
+		/// Gets character's agility (DEX).
+		/// </summary>
+		public int Dex { get { return this.DexByJob + this.DexInvested + this.DexBonus; } }
+
+		/// <summary>
+		/// Character's session objects.
+		/// </summary>
+		public SessionObjects SessionObjects { get; } = new SessionObjects();
+
+		/// <summary>
+		/// Character's skills.
+		/// </summary>
+		public Skills Skills { get; }
+
+		/// <summary>
+		/// Character's abilities.
+		/// </summary>
+		public Abilities Abilities { get; }
+
+		/// <summary>
+		/// Character's properties.
+		/// </summary>
+		/// <remarks>
+		/// Beware, some of these are reference properties, that can't be
+		/// set directly. Use this object's actual properties whenever
+		/// possible.
+		/// </remarks>
+		public Properties Properties { get; } = new Properties();
+
+		/// <summary>
+		/// Returns minimum physical ATK.
+		/// </summary>
+		public int MinPAtk
```

Comment:
Are these all based on the client files now?

Reply:
Yes